src/ssm.js

```diff
@@ -100,8 +103,14 @@
         fireAllMethodsInArray(leaveMethods);
         fireAllMethodsInArray(enterMethods);
         fireAllMethodsInArray(resizeMethods);
+
+        if (changed) {
```

Comment:
Ah, yes... it seems you're right. The idea was to have it run just once regardless of which event was triggered. I've pushed another update for this.